### PR TITLE
Fix to detect binary stl file format when header starts with "solid".

### PR DESCRIPTION
**Pull request by dreammaker**
Addresses #56.

### CODE REVIEW

File: src/Files.jl [hunk 1]
```diff
@@ -87,4 +85,41 @@ function mesh(path::String; format=:autodetect)
...
+    seekend(io)
+    len = position(io)
+    seekstart(io)
+    len < 80 && return false
+    header = readbytes(io, 80) # skip header
+    seekstart(io)
+    return header[1:6] == b"solid " && !detect_stlbinary(io)
+end
+
+function detect_stlbinary(io)
```
Comment (sjkelly) [inline src/Files.jl:99]: Can you document this? The approach is not readily apparent to me.

Comment (dreammaker) [inline src/Files.jl:99]: I copied the bulk of it from Simon's branch [here](https://github.com/JuliaIO/FileIO.jl/blob/sd/improvements/src/registry.jl#L153).  But I can definitely document it.

Comment (sjkelly) [inline src/Files.jl:99]: Don't worry about it. It should go into FileIO. We are going to nuke `Meshes.Files` hopefully this weekend anyway.

Comment (dreammaker) [inline src/Files.jl:99]: Then if you don't mind, I'll merge this, or at least the one on the v0.1.x branch and cut a release.

Comment (sjkelly) [inline src/Files.jl:99]: Let's close this one and put it on 0.1.x. We need to fix the 0.4 failure first.

**Pull request closed without merging**